Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -0,0 +1,1301 @@
+        }
+
+        /// <inheritdoc/>
+        public bool IsPrivateAddressRange(IPObject address)
+        {
+            if (address == null)
+            {
+                throw new ArgumentNullException(nameof(address));
+            }
+
+            // See conversation at https://github.com/jellyfin/jellyfin/pull/3515.
+            if (TrustAllIP6Interfaces && address.AddressFamily == AddressFamily.InterNetworkV6)
+            {
+                return true;
+            }
+            else
+            {
+                return address.IsPrivateAddressRange();
+            }
+        }
+
+        /// <inheritdoc/>
+        public bool IsExcludedInterface(IPAddress address)
+        {
+            return _bindExclusions.ContainsAddress(address);
+        }
+
+        /// <inheritdoc/>
+        public Collection<IPObject> GetFilteredLANSubnets(Collection<IPObject>? filter = null)
+        {
+            if (filter == null)
+            {
+                return _lanSubnets.Exclude(_excludedSubnets).AsNetworks();
+            }
+
+            return _lanSubnets.Exclude(filter);
+        }
+
+        /// <inheritdoc/>
+        public bool IsValidInterfaceAddress(IPAddress address)
+        {
+            return _interfaceAddresses.ContainsAddress(address);
+        }
+
+        /// <inheritdoc/>
+        public bool TryParseInterface(string token, out Collection<IPObject>? result)
+        {
+            result = null;
+            if (string.IsNullOrEmpty(token))
+            {
+                return false;
+            }
+
+            if (_interfaceNames != null && _interfaceNames.TryGetValue(token.ToLower(CultureInfo.InvariantCulture), out int index))
+            {
+                result = new Collection<IPObject>();
+
+                _logger.LogInformation("Interface {0} used in settings. Using its interface addresses.", token);
+
+                // Replace interface tags with the interface IP's.
+                foreach (IPNetAddress iface in _interfaceAddresses)
+                {
+                    if (Math.Abs(iface.Tag) == index
+                        && ((IsIP4Enabled && iface.Address.AddressFamily == AddressFamily.InterNetwork)
+                            || (IsIP6Enabled && iface.Address.AddressFamily == AddressFamily.InterNetworkV6)))
+                    {
+                        result.AddItem(iface);
+                    }
+                }
+
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Reloads all settings and re-initialises the instance.
+        /// </summary>
+        /// <param name="configuration">The <see cref="NetworkConfiguration"/> to use.</param>
+        public void UpdateSettings(NetworkConfiguration configuration)
+        {
+            NetworkConfiguration config = (NetworkConfiguration)configuration ?? throw new ArgumentNullException(nameof(configuration));
```

Comment:
Was picked up before - so i changed it to NetworkConfiguration. Thought i'd changed it back (have done now). The param is an object, as once the INetworkManager has moved to the correct folder, the interface cannot see the NetworkConfiguration object - and you get a circular link if you try to add.

Reply:
Sounds like something that should be fixed

Reply:
It's because I moved INetworkManager back to it's current location - to reduce the number of changes in the uses clauses.

Which leaves two options - 

1) PR to move INetworkManager back into Jellyfin.Networking and updates the uses.
2) PR to move the config into MediaBrowser.Common.

Once either is done - this can be changed back from an object, It's on the TODO list.

I prefer the first, as it keeps all the network stuff together.

Reply:
Isn't INetworkManager  already in Jellyfin.Networking?

We currently have some config classes in MediaBrowser.Model.Configuration and MediaBrowser.Common.Configuration.

Reply:
Yes - but only so that this PR can be vetted and build without doing anything. (In this PR old networkManager is still in use).

In PR2 it's moved, overwriting the existing one, and old networkManager is removed.

This was done in an attempt to reduce the sizes of the PR (that really worked!!!)